src/resources/js/dialog.js

```diff
@@ -1,49 +1,71 @@
 var tribe = tribe || {};
 tribe.dialogs = tribe.dialogs || {};
-tribe.dialogs.dialogs = tribe.dialogs.dialogs || {};
-tribe.dialogs.events = tribe.dialogs.events || {};
-
-	( function ( obj ) {
-		'use strict';
-
-		document.addEventListener(
-			'DOMContentLoaded',
-			function () {
-				tribe.dialogs.dialogs.forEach(function(dialog) {
-					var objName     = 'dialog_obj_' + dialog.id;
-					window[objName] = new window.A11yDialog({
-						appendTarget: dialog.appendTarget,
-						bodyLock: dialog.bodyLock,
-						closeButtonAriaLabel: dialog.closeButtonAriaLabel,
-						closeButtonClasses: dialog.closeButtonClasses,
-						contentClasses: dialog.contentClasses,
-						effect: dialog.effect,
-						effectEasing: dialog.effectEasing,
-						effectSpeed: dialog.effectSpeed,
-						overlayClasses: dialog.overlayClasses,
-						overlayClickCloses: dialog.overlayClickCloses,
-						trigger: dialog.trigger,
-						wrapperClasses: dialog.wrapperClasses
-					});
-
-					window[objName].on('show', function (dialogEl, event) {
-						if ( event ) {
-							event.preventDefault();
-							event.stopPropagation();
-						}
-
-						jQuery( tribe.dialogs.events ).trigger( dialog.showEvent, [dialogEl, event] );
-					});
-
-					window[objName].on('hide', function (dialogEl, event) {
-						if ( event ) {
-							event.preventDefault();
-							event.stopPropagation();
-						}
-
-						jQuery( tribe.dialogs.events ).trigger( dialog.closeEvent, [dialogEl, event] );
-					});
-				});
-			}
-		)
-	})(tribe.dialog);
+
+( function( $, obj ) {
+	'use strict';
+
+	var $document = $( document );
+	tribe.dialogs.dialogs = tribe.dialogs.dialogs || [];
+	tribe.dialogs.events = tribe.dialogs.events || {};
+
+	/**
+	 * Get the dialog name.
+	 *
+	 * @since TBD
+	 *
+	 * @param {obj} dialog The dialog object
+	 *
+	 * @return {string} the dialog name.
+	 */
+	obj.getDialogName = function( dialog ) {
+		return 'dialog_obj_' + dialog.id;
```

Comment:
Should we be prefixing this?

Reply:
I feel we're moving far away from the original fix which was just replacing:

`tribe.dialogs.dialogs = tribe.dialogs.dialogs || {};`
with this:
`tribe.dialogs.dialogs = tribe.dialogs.dialogs || [];`

And we're reimplementing the dialogs, with the QA risk implications, given that I was not the one who implemented this. Maybe we can end up handling a re-implementation on a separate ticket or modifying the ticket so QA knows that they need to test not only the original thing but the entire dialogs.

Reply:
Agree here